meetings/working-groups/params-improvements/PI-2022-11-03.md

```diff
@@ -0,0 +1,125 @@
+# Params Improvements Working Group Meeting for November 3, 2022
+The second meeting of the _`params` improvements_ working group was focused on the code generated for arguments to `params ReadOnlySpan<T>`. 
+
+## Span allocation
+The compiler _may allocate a span on the stack_ when:
+- the span is implicitly allocated (as a `params` argument or as a _collection literal_),
+- the span length is known at compile time, and
+- the compiler can determine _through escape analysis_ that no references to the span escape the containing method.
+
+For arguments to a `params ReadOnlySpan<T>` parameter (which is implicitly `scoped`), the conditions above are satisfied.
+
+We have a request with the runtime team to support _fixed size arrays_ of managed types, at least for fields.
+If we have _fixed size array_ fields, we can define `struct` types with inline arrays and use locals for stack allocated arrays.
+See [#61135](https://github.com/dotnet/runtime/issues/61135).
+
+For example, consider a `FixedSizeArray2` type defined below which includes an inline 2 element array:
+```csharp
+struct FixedSizeArray2<T>
+{
+    public T[2] Array; // pseudo-code for inline array
+}
+```
+
+With that type, a call to `WriteLine("{0}, {1}", x, y)` could be emitted as:
+```csharp
+var _tmp1 = new FixedSizeArray2<object>();
+_tmp1.Array[0] = x;
+_tmp1.Array[1] = y;
+var _tmp2 = new ReadOnlySpan<object>(_tmp.Array);
+
+// WriteLine(string format, params ReadOnlySpan<object> args);
+WriteLine("{0}, {1}", _tmp2);
+```
+
+Ideally the BCL will provide types such as `FixedSizeArray1`, `FixedSizeArray2`, etc. for a limited number of array lengths.
+And if the compilation requires spans for other argument lengths, the compiler will generate and emit the additional types.
+
+The number of arguments passed to a `params` parameter is _not considered_ when determining whether to implicitly allocate the span on the stack.
+To avoid implicit stack allocation at a call site, the calling code should allocate the array explicitly with `new[] { ... }`.
```

Comment:
I would def add a comment that we believe this is acceptable, but htat we will be observing the fallout of this to see if it causes problems for the ecosystem on source-update.